apollo-router-core/src/services/router_service.rs

```diff
@@ -308,14 +309,18 @@ impl PluggableRouterServiceBuilder {
         let (router_service, router_worker) = Buffer::pair(
             ServiceBuilder::new().service(
```

Comment:
APQ layer should be added here:
`ServiceBuilder::new().layer(APQLayer::default()).service(`
That way the layer is executed first and user layers come after.

Reply:
fixed in c42d9f9